### PR TITLE
SJQ2T-196: BE Endpoint for View Buckets

### PR DESCRIPTION
**Pull request by DarkoVeljanoskiHtec**
endpoint, service and test for view bucket functionality.

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java [hunk 1]
```diff
@@ -97,4 +98,60 @@ public ResponseEntity<BucketResponse> getBucket(@PathVariable UUID uuid) {
...
+                            "[{\"id\": \"a6fd6d95-0a60-43ff-961f-2b9b2ff72f95\","
+                                + "\"name\": \"Name\","
+                                + "\"companyId\": \"a6fd6d95-0a60-43ff-961f-2b9b2ff72f95\","
+                                + "\"description\": \"Description\","
+                                + "\"size\": \"1000\"}},"
+                                + "{\"id\": \"u7yd987h-0a79-42dd-961s-7sfh564kdv2s\","
+                                + "\"name\": \"Name\","
+                                + "\"companyId\": \"a6fd6d95-0a60-43ff-961f-2b9b2ff72f95\","
+                                + "\"description\": \"Description\","
+                                + "\"size\": \"1000\"}]")
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java]: two examples objects that are exactly the same?

File: backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java [hunk 2]
```diff
@@ -97,4 +98,60 @@ public ResponseEntity<BucketResponse> getBucket(@PathVariable UUID uuid) {
         new ResponseEntity<>(bucketResponse, HttpStatus.OK);
     return bucketResponseEntity;
   }
+
+  @Operation(summary = "Get all buckets")
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java:173]: should this get all buckets just for the time being, or should we only get the user's private buckets as well as all public buckets from the user's company?

File: backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java [hunk 3]
```diff
@@ -25,4 +27,12 @@ public BucketResponse getBucket(UUID uuid) {
     BucketResponse bucketResponse = modelMapper.map(bucketEntity, BucketResponse.class);
     return bucketResponse;
   }
+
+  @Override
+  public List<BucketResponse> listAllBuckets() {
+    List<BucketEntity> entityList = (List<BucketEntity>) bucketsRepository.findAll();
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java]: findAll() returns an Iterable? I can't believe this works

nice

File: backend/src/test/java/com/htecgroup/skynest/service/impl/BucketServiceImplTest.java [hunk 4]
```diff
@@ -25,6 +27,21 @@ class BucketServiceImplTest {
   @Spy private ModelMapper modelMapper;
   @Spy @InjectMocks private BucketServiceImpl bucketService;
 
+  @Test
+  void listAllBuckets() {
+    List<BucketEntity> bucketEntityList = new ArrayList<>();
+    bucketEntityList.add(BucketEntityUtil.getPrivateBucket());
```
Comment (ivanfhtec) [inline backend/src/test/java/com/htecgroup/skynest/service/impl/BucketServiceImplTest.java]: Collections.singletonList()?

File: backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java (hunk 1 above)
Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java]: I forgot to change it

File: backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java (hunk 3 above)
Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java]: Yes, it works like this :)

File: backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java (hunk 2 above)
Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java:173]: I don't know, I thought we're going to do that later with the permissions, but we could talk it out.

**Review by ivanfhtec — APPROVED** [review]

**Pull request merged**